packages/synapse-react-client/src/mocks/faker/generateFakeEntity.ts

```diff
@@ -1,26 +1,54 @@
 import { faker } from '@faker-js/faker'
-import { Entity, Project } from '@sage-bionetworks/synapse-types'
+import { Entity, EntityHeader, Project } from '@sage-bionetworks/synapse-types'
 import { pickRandomMockUser } from './fakerUtils'
+import { MockEntityData } from '../entity/MockEntityData'
 
-function generateBaseEntity(overrides?: Partial<Entity>): Entity {
-  return {
-    id: `syn${faker.number.int({ min: 10000, max: 99999 })}`,
+export function generateBaseEntity(
+  entityDataOverrides?: Partial<Entity>,
+  idOverride?: number,
+): MockEntityData {
+  const id = idOverride ?? faker.number.int({ min: 10000, max: 99999 })
+  const entity = {
+    id: `syn${id}`,
     createdBy: String(pickRandomMockUser().id),
     createdOn: faker.date.anytime().toISOString(),
     etag: faker.string.uuid(),
     modifiedBy: String(pickRandomMockUser().id),
     modifiedOn: faker.date.anytime().toISOString(),
     name: faker.lorem.words({ min: 1, max: 4 }),
     concreteType: 'org.sagebionetworks.repo.model.FileEntity',
-    ...overrides,
+    ...entityDataOverrides,
+  } satisfies Entity
+  const header: EntityHeader = {
+    id: entity.id,
+    name: entity.name,
+    type: entity.concreteType,
+    createdBy: entity.createdBy,
+    createdOn: entity.createdOn,
+    modifiedBy: entity.modifiedBy,
+    modifiedOn: entity.modifiedOn,
+    benefactorId: id,
+    isLatestVersion: true,
+    versionLabel:
+      'versionLabel' in entity ? (entity.versionLabel as string) : undefined,
+    versionNumber:
+      'versionNumber' in entity ? (entity.versionNumber as number) : undefined,
   }
-}
 
-export function generateProject(overrides?: Partial<Project>): Project {
   return {
-    ...generateBaseEntity(overrides),
+    id: entity.id!,
+    entity: entity,
+    name: entity.name,
+    entityHeader: header,
+  }
+}
```

Comment:
Updated to return the bundle of mock data, including entity headers, that we pass along to mock service worker, so that these are easier to use in those tests and stories.